packages/tldraw/src/lib/ui/components/primitives/layout.tsx

```diff
@@ -1,10 +1,55 @@
 import classNames from 'classnames'
 import { Slot } from 'radix-ui'
-import { HTMLAttributes, ReactNode, forwardRef } from 'react'
+import { HTMLAttributes, ReactNode, createContext, forwardRef, useContext } from 'react'
+
+/** @public */
+export interface TldrawUiOrientationContext {
+	orientation: 'horizontal' | 'vertical'
+	tooltipSide: 'top' | 'right' | 'bottom' | 'left'
+}
+
+const TldrawUiOrientationContext = createContext<TldrawUiOrientationContext>({
+	orientation: 'horizontal',
+	tooltipSide: 'bottom',
+})
+
+/** @public */
+export interface TldrawUiOrientationProviderProps {
+	children: ReactNode
+	orientation: 'horizontal' | 'vertical'
+	tooltipSide?: 'top' | 'right' | 'bottom' | 'left'
+}
+/** @public @react */
+export function TldrawUiOrientationProvider({
+	children,
+	orientation,
+	tooltipSide,
+}: TldrawUiOrientationProviderProps) {
+	const prevContext = useTldrawUiOrientation()
+	const tooltipSideToUse =
```

Comment:
Not sure I understand why we need to do this?

Reply:
I've added a comment here trying to explain this.

Essentially, the default tooltip side is almost always fine - but when you do change it, you then want it to cascade down until you explicitly change it again. However, if you then switch orientation your tooltip side will be wrong - it'll be covering items in your layout. So in that case, we reset it to a default.